framework/architecture/hibernate/listeners.rst

```diff
@@ -0,0 +1,129 @@
+immediately after a certain entity operation (e.g. ``EntityManager#create()`` or ``Entity#setValue()``) and therefore
+this listener is independent of any hibernate action.
+
+Normally entity facade listeners are registered as hivemind services which then are injected into the
+:java:ref:`EntityFacadeListenerManagerImpl<ch.tocco.nice2.persist.hibernate.listener.EntityFacadeListenerManagerImpl>`,
+which manages all listeners and can also be used to fire events.
+
+It is also possible to temporarily register an :java:ref:`EntityFacadeListener<ch.tocco.nice2.persist.entity.events.EntityFacadeListener>` for the current context
+using ``Context#addEntityFacadeListener()``. These listeners will be called for every entity and will only be called
+for events of the context they belong to.
+Likewise it is possible to add a temporary listener using ``EntityManager#addEntityFacadeListener()`` which will only
+be called for events of this entity manager.
+
+When an event is fired, the :java:ref:`EntityFacadeListenerManagerImpl<ch.tocco.nice2.persist.hibernate.listener.EntityFacadeListenerManagerImpl>`
+searches for all registered hivemind listeners (that accept events of the affected entity) as well as all listeners of the current context
+and entity manager.
+
+``entityCreating`` event
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+This event is fired directly after a new entity instance is created (no fields are set at this point) by the
+:java:ref:`EntityFactoryImpl<ch.tocco.nice2.persist.hibernate.pojo.EntityFactoryImpl>`. Because all entity instances
+are created by the entity factory (even those that are loaded from the database), the event is only fired when the
+primary key is not set yet.
+
+``entityDeleting`` event
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+This event is fired when ``Entity#delete()`` is called.
+Note that the entity won't be deleted from the database until the transaction is committed.
+
+``entityChanging`` event
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+This event is fired every time a field is updated (i.e. if ``Entity#setValue()`` or the setter for this field is called).
+This is done in ``AbstractHibernateEntity#internalSetValue()`` which is called by both ``Entity#setValue()`` and the
+generated setter method.
+
+``entityRelationChanging`` event
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This event is fired every time a relation is modified (i.e. if an entity is added to or removed from a relation).
+This is done in ``AbstractRelationAdapter#fireRelationChangingEvent()`` which in turn is called by the
+:java:ref:`ToOneRelationAdapter<ch.tocco.nice2.persist.hibernate.pojo.relation.ToOneRelationAdapter>` and the
+:java:ref:`ToManyRelationAdapter<ch.tocco.nice2.persist.hibernate.pojo.relation.ToManyRelationAdapter>`.
+An event is also always fired for the reverse side of the relation.
+
+See the chapter :doc:`abstract-pojo-entity` for a description when the ``adjusting`` flag is set to true.
+
+``entityReceivedValues`` event
```

Comment:
> \`\`entityReceivedValues\`\` event

I'd add a `… (deprecated)` to the title, so people don't need to read on when writing new code.